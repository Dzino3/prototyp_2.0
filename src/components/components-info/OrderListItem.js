import React, { useEffect, useState } from 'react'
import '../style/listStyle.css'
import OrderListItemSourced from './OrderListItemSourced';
import OrderListItemUnsourced from './OrderListItemUnsourced';
import OrderListItemInventory from './OrderListItemInventory';
import OrderListItemCanceled from './OrderListItemCanceled';

function OrderListItem({ item, selec = false, t }) {

  const [selected, setSelected] = useState(selec);


  const toggle = () => {
    setSelected(!selected)
  }

  const getStatusComponent = () => {
    switch (item.PROCESS_STATUS) {
      case 'sourced':
        return <OrderListItemSourced item={item} t={t}></OrderListItemSourced>;
      case 'unsourced':
        return <OrderListItemUnsourced item={item} t={t}></OrderListItemUnsourced>;
      case 'inventory':
        return <OrderListItemInventory item={item} t={t}></OrderListItemInventory>;
      case 'canceled':
        return <OrderListItemCanceled item={item} t={t}></OrderListItemCanceled>;
    }
  }
  return (

    <div className='item'>
      <div className='beforeDiv' onClick={() => toggle()}>
        <div className='orderDiv1'>
          <div className={item.ORDER_NUMBER === '654321' ? 'orderImg2' : 'orderImg'}></div>

          <div className="topRow">
            <h3 className='listItemH3'>{t('listItemLabel1')} {item.ORDER_NUMBER}</h3>
            <div className='mobilStatusHidden'>
              <div className='orderStatusDiv'>
                <p className={'orderStatus' + item.PROCESS_STATUS}>{item.PROCESS_STATUS}</p>
              </div>
              <span className='toggleSpan'>{selected === true ? '▼' : '◄'}</span>
            </div>
          </div>

          <div className='orderDiv2'>
            <div className='orderListDivFlex'>
              <p className='p1'>{t('listItemLabel2')}: </p>
              <p>{item.BASE_CODE_DESCR} ({item.BASE_CODE})</p>
            </div>
          </div>
        </div>
        
        <div className='mobilStatus'>
          <div className='orderStatusDiv'>
            <p className={'orderStatus' + item.PROCESS_STATUS}>{item.PROCESS_STATUS}</p>
          </div>
          <span className='toggleSpan'>{selected === true ? '▼' : '◄'}</span>
        </div>
      </div>

      <div>
        <div className={selected === true ? 'afterDivshow' : 'afterDiv'}>
          <div className='afterDiv1'>
            {getStatusComponent()}
          </div>
        </div>
      </div>
    </div>

  )
}

export default OrderListItem