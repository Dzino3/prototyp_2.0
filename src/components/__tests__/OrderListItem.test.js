import React from "react";
import renderer from "react-test-renderer";
import OrderListItem from "../components-info/OrderListItem";

import { useState as useStateMock } from "react";

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}))
describe(`Rendering OrderListItem`, () => {

    const setState = jest.fn()

    beforeEach(() => {
      useStateMock.mockImplementation(init => [init, setState])
    })

    const item = {
        PROCESS_STATUS: "Shipped"
    }

  test("should render OrderListItem", function () {
    let tree = renderer.create(<OrderListItem item={item} t={key => key}/>);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('orderListItem state', () => {
      
    let tree = renderer.create(
      <OrderListItem item={item} t={key => key}/>
    );
    expect(setState).toHaveBeenCalledTimes(0)
    expect(tree).toBeTruthy()
   
  })
});
