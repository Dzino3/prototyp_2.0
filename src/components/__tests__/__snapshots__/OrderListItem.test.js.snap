// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering OrderListItem should render OrderListItem 1`] = `
<div
  className="item"
>
  <div
    className="beforeDiv"
    onClick={[Function]}
  >
    <div
      className="orderDiv1"
    >
      <div
        className="topRow"
      >
        <div
          className="mobilStatusHidden"
        >
          <div
            className="orderStatusDiv"
          >
            <p
              className="orderStatusShipped"
            >
              Shipped
            </p>
          </div>
          <span
            className="toggleSpan"
          >
            ◄
          </span>
        </div>
      </div>
      <div
        className="orderDiv2"
      >
        <div
          className="orderImg2"
        />
        <div
          className="listH3andPBlock"
        >
          <h3
            className="listItemH3"
          >
            listItemLabel1
             
          </h3>
          <div
            className="orderListDivFlex"
          >
            <p
              className="p1"
            >
              listItemLabel2
              : 
            </p>
            <p>
               (
              )
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mobilStatus"
    >
      <div
        className="orderStatusDiv"
      >
        <p
          className="orderStatusShipped"
        >
          Shipped
        </p>
      </div>
      <span
        className="toggleSpan"
      >
        ◄
      </span>
    </div>
  </div>
  <div>
    <div
      className="afterDiv"
    >
      <div
        className="afterDiv1"
      >
        <div
          className="afterDiv1-2"
        >
          <div
            className="afterDiv1-3"
          >
            <div
              className="contactUs"
            >
              <p
                className="needHelp"
              >
                contactUsText
              </p>
              <a
                href="mailto:codux@gmx.de?subject=Order #undefined&body=Hello John Deere Support, I need help with this order:"
              >
                <input
                  className="contactUsButton"
                  type="button"
                  value="contactUsButton"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
